components/TextFields/tests/snapshot/MDCTextFieldFilledControllerLeadingImageArabicSnapshotTests.m

```diff
@@ -0,0 +1,55 @@
+// Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MDCAbstractTextFieldSnapshotTests+I18N.h"
+#import "MDCAbstractTextFieldSnapshotTests+LeadingImage.h"
+#import "MDCAbstractTextFieldSnapshotTests.h"
+#import "MaterialTextFields.h"
+
+@interface MDCTextFieldFilledControllerLeadingImageArabicSnapshotTests
+    : MDCAbstractTextFieldSnapshotTests <MDCTextFieldSnapshotTestCaseHooking>
+@end
+
+@implementation MDCTextFieldFilledControllerLeadingImageArabicSnapshotTests
+
+- (void)setUp {
+  [super setUp];
+
+  // Uncomment below to recreate the golden images for all test methods. Add it to a test method to
+  // update only that golden image.
+  //  self.recordMode = YES;
+
+  self.textField.clearButtonMode = UITextFieldViewModeAlways;
```

Comment:
Tip: Does it make sense to include some screenshot tests with the clear button off?

Reply:
We could certainly add some, but I figured the clear button alignment/position is a more "complex" layout than without.  As a result, the tests I wrote so far tend to turn it on.